common/src/main/java/com/wynntils/screens/itemfilter/widgets/NumericValueWidget.java

```diff
@@ -0,0 +1,320 @@
+/*
+ * Copyright © Wynntils 2024.
+ * This file is released under LGPLv3. See LICENSE for full license details.
+ */
+package com.wynntils.screens.itemfilter.widgets;
+
+import com.mojang.blaze3d.vertex.PoseStack;
+import com.wynntils.core.WynntilsMod;
+import com.wynntils.core.text.StyledText;
+import com.wynntils.screens.base.widgets.TextInputBoxWidget;
+import com.wynntils.screens.base.widgets.WynntilsCheckbox;
+import com.wynntils.screens.itemfilter.ItemFilterScreen;
+import com.wynntils.services.itemfilter.filters.AnyStatFilters;
+import com.wynntils.services.itemfilter.filters.RangedStatFilters;
+import com.wynntils.services.itemfilter.type.ItemStatProvider;
+import com.wynntils.services.itemfilter.type.StatFilter;
+import com.wynntils.services.itemfilter.type.StatProviderAndFilterPair;
+import com.wynntils.utils.colors.CommonColors;
+import com.wynntils.utils.render.FontRenderer;
+import com.wynntils.utils.render.type.HorizontalAlignment;
+import com.wynntils.utils.render.type.TextShadow;
+import com.wynntils.utils.render.type.VerticalAlignment;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import net.minecraft.client.gui.GuiGraphics;
+import net.minecraft.client.gui.components.Button;
+import net.minecraft.network.chat.Component;
+import org.lwjgl.glfw.GLFW;
+
+public abstract class NumericValueWidget<T> extends GeneralValueWidget {
+    protected final Button greaterThanButton;
+    protected final Button lessThanButton;
+    protected final TextInputBoxWidget singleInput;
+    protected final TextInputBoxWidget rangedMinInput;
+    protected final TextInputBoxWidget rangedMaxInput;
+    protected final TextInputBoxWidget greaterThanInput;
+    protected final TextInputBoxWidget lessThanInput;
+    protected final WynntilsCheckbox allCheckbox;
+
+    protected boolean greaterThanEqual = false;
+    protected boolean lessThanEqual = false;
+
+    protected NumericValueWidget(
+            List<StatProviderAndFilterPair> filters,
+            ItemStatProvider<?> itemStatProvider,
+            ItemFilterScreen filterScreen) {
+        super(Component.literal("Integer Value Widget"), itemStatProvider, filterScreen);
```

Comment:
```suggestion
        super(Component.literal("Numeric Value Widget"), itemStatProvider, filterScreen);
```

Or add title as a parameter, either works